project11/project11.sql

```diff
@@ -0,0 +1,248 @@
+
+-- Функции объектов
+CREATE TYPE function AS ENUM ('Жилое помещение', 'Кафе', 'Спорт. площадка', 'Бассейн', 'Штаб делегации', 'Столовая');
+
+-- Олимпийские объекты:
+--  id SERIAL - ключ для внешних ссылок
+--  address - адрес объекта (адрес до дома/строения)
+--  function - функция объекта (ссылается на список известных функций)
+--  name - опциональное имя объекта
+--  По одному адресу не более одного объекта с одинаковой функцией
+
+CREATE TABLE object_ (
+    id SERIAL, 
+    address TEXT NOT NULL,
+    function FUNCTION NOT NULL, 
+    name TEXT NOT NULL,
+
+    -- Ключи
+	PRIMARY KEY(id)
+	-- Других ключей нет
+);
+
+-- Суррогатным основным ключом выступает id (для организации внешних ссылок)
+-- Из одной страны может быть только одна делегация
+-- Адрес делегации должен быть указан
+-- Сейчас может возникнуть неконсисентность, что house_id оказался не id офиса делегации.
+CREATE TABLE delegation(
+    id SERIAL PRIMARY KEY,
+    -- У одной страны только одна национальная делегация
+	country TEXT NOT NULL UNIQUE,
```

Comment:
Probably it's worth having an ENUM for countries.

Reply:
done